app/Services/Bibles/BibleFilesetService.php

```diff
@@ -158,4 +159,70 @@ public static function getVerseTextFilterBy(
             ->get()
             ->implode('verse_text', ' ');
     }
+
+
+    /**
+     * Match filesets with text filesets based on size code.
+     *
+     * This method iterates over a collection of filesets and matches them with corresponding text filesets
+     * from a separate collection. The matching is based on the size code of the filesets. It looks for text
+     * filesets whose size code is either identical to or contains the size code of the current fileset.
+     *
+     * @param \Illuminate\Support\Collection $filesets The collection of filesets to be matched.
+     * @param \Illuminate\Support\Collection $text_filesets The collection of text filesets to match against.
+     * @param \Illuminate\Support\Collection $bible_hashes A collection mapping fileset hash IDs to bible IDs.
+     *
+     * @return array An associative array where the keys are fileset IDs and the values are arrays of matching text filesets.
+     *
+     */
+    public static function matchFilesetsByTextSize(
+        Collection $filesets,
+        Collection $text_filesets,
+        Collection $bible_hashes
+    ) : Array {
+        $fileset_text_info = [];
+        foreach ($filesets as $fileset) {
+            $bibleId = $bible_hashes[$fileset->hash_id] ?? null;
+            if ($bibleId && isset($text_filesets[$bibleId])) {
+                foreach ($text_filesets[$bibleId] as $plain_fileset) {
+                    // Check if the set size code of the text fileset matches the current fileset's size code
+                    // or if the text fileset's size code contains the current fileset's size code.
+                    // E.g fileset = NT and text plain = NT OR text plain = NTP
+                    if ($plain_fileset->set_size_code === $fileset->set_size_code ||
+                        str_contains($plain_fileset->set_size_code, $fileset->set_size_code)
+                    ) {
+                        $fileset_text_info[$fileset->id][] = $plain_fileset;
+                    }
+                }
+            }
+        }
+
+        return $fileset_text_info;
+    }
+
+    public static function getFilesetsByIds(Collection $ids) : Collection
+    {
+        return BibleFileset::select(['hash_id', 'id', 'set_size_code'])
```

Comment:
@bradflood , This query is considering now that the audio fileset belonging to the playlist item should have the same `set_size_code` as the plain text fileset